ml_peg/app/utils/build_callbacks.py

```diff
@@ -221,6 +223,160 @@ def show_struct(active_cell):
+    scatter_meta_store_id: str,
+    last_cell_store_id: str,
+    column_handlers: dict[str, Callable[[str, str], tuple[Component, dict] | None]],
+    default_handler: Callable[[str, str], tuple[Component, dict] | None] | None = None,
+    refresh_message: str = "Click on a metric to view scatter plots.",
+    model_key: str = "MLIP",
+) -> None:
+    """
+    Render scatter content and persist model-specific metadata for asset callbacks.
+
+    Parameters
+    ----------
+    table_id
+        Dash table identifier emitting ``active_cell`` callbacks.
+    table_data
+        Pre-rendered table rows used to look up the clicked model.
+    plot_container_id
+        Div ID hosting the rendered plot content.
+    scatter_meta_store_id
+        Store component ID that tracks the latest plot metadata.
+    last_cell_store_id
+        Store component ID used to reset when the same cell is clicked twice.
+    column_handlers
+        Mapping of column identifiers to callables returning ``(content, meta)``.
+    default_handler
+        Fallback callable invoked when ``column_handlers`` has no entry.
+    refresh_message
+        Message displayed when a user re-clicks the active cell.
+    model_key
+        Key in ``table_data`` used to look up the model display name.
+    """
+
+    @callback(
+        Output(plot_container_id, "children"),
+        Output(scatter_meta_store_id, "data"),
+        Output(last_cell_store_id, "data"),
+        Input(table_id, "active_cell"),
+        State(last_cell_store_id, "data"),
+        prevent_initial_call=True,
+    )
+    def _update_scatter_and_meta(active_cell, last_cell):
+        """
+        Map table cells to plot content.
+
+        Parameters
+        ----------
+        active_cell
+            Dash ``active_cell`` data dict from the metrics table.
+        last_cell
+            Previously clicked cell stored in ``last_cell_store_id``.
+
+        Returns
+        -------
+        tuple
+            Plot container children, scatter meta, and new ``last_cell`` value.
+        """
+        if not active_cell:
             raise PreventUpdate
-        raise ValueError("Invalid column_id")
+        if last_cell and last_cell == active_cell:
+            return html.Div(refresh_message), None, None
+        row = active_cell.get("row")
+        column = active_cell.get("column_id")
+        if row is None or column is None or row < 0 or row >= len(table_data):
+            raise PreventUpdate
+        model_display = table_data[row].get(model_key)
+        if not model_display:
+            raise PreventUpdate
+        handler = column_handlers.get(column)
+        if handler is None:
+            handler = default_handler
+        if handler is None:
+            raise PreventUpdate
+        result = handler(model_display, column)
+        if not result:
+            raise PreventUpdate
+        content, meta = result
+        return content, meta, active_cell
+
+
+def model_asset_from_scatter(
```

Comment:
Where does the model-specific part come in?

As above, I also wonder if asset it the most useful description? Is this actually loading from `/assets`?